src/sas/qtgui/Plotting/Plotter.py

```diff
@@ -280,9 +280,27 @@ def plot(self, data=None, color=None, marker=None, hide_error=False, transform=T
             x_range = self.setRange.xrange()
             y_range = self.setRange.yrange()
         else:
-            # Use default ranges given by matplotlib
-            x_range = default_x_range = self.ax.get_xlim()
-            y_range = default_y_range = self.ax.get_ylim()
+            if isinstance(data, Data1D):
+                # Get default ranges from data
+                # factors of .99 and 1.01 provides a small gap so end points not shown right at edge
+                default_x_range = (0.99*np.min(x), 1.01*np.max(x))
```

Comment:
Multiple occurences of these two parameters `0.99` and `1.09` would suggest replacing them with predefined (even local) constants.